omni-relayer/src/startup/eth.rs

```diff
@@ -0,0 +1,48 @@
+use std::sync::Arc;
+
+use anyhow::{Context, Result};
+use tokio_stream::StreamExt;
+
+use alloy::{
+    primitives::Address,
+    providers::{Provider, ProviderBuilder, WsConnect},
+    rpc::types::Filter,
+};
+
+use nep141_connector::Nep141Connector;
+
+use crate::{defaults, utils};
+
+pub async fn start_indexer(connector: Arc<Nep141Connector>) -> Result<()> {
+    let http_provider = ProviderBuilder::new().on_http(
+        defaults::ETH_RPC_MAINNET
+            .parse()
+            .context("Failed to parse ETH rpc provider as url")?,
+    );
+
+    let ws_provider = ProviderBuilder::new()
+        .on_ws(WsConnect::new(defaults::ETH_WS_MAINNET))
+        .await
+        .context("Failed to initialize WS provider")?;
+
+    let latest_block = http_provider.get_block_number().await?;
+    let from_block = latest_block.saturating_sub(1000);
```

Comment:
Let's store the last processed block in the Redis db

Reply:
It's easy to do with near lake indexer, but I believe that we can't track each block on eth blockchain. We only are able to handle blocks that are already filtered out (aka only blocks with events on our bridge account). Is it good enough or it's better to dig a bit and try to update db on each block?

Reply:
Instead of tweaking eth event tracker, we can modify "continue from last processed block" algorithm. I think there's no point to start indexer from block that it's older than X days (I guess X should be 3 or even 1), because person most likely wouldn't expect relayer to help him with finalizing transactions after a day of sending it. So we probably can update last processed block in redis on every upcomming event on bridge token factory and the finished algo will be something like:
```pseudocode
if final_block - redis.get("eth_last_processed_block") > avg_blocks_produced_per_day * X then
  start indexer from final_block - 10_000
else
  start indexer from redis.get("eth_last_processed_block")
```

Reply:
I think  `redis.get("eth_last_processed_block").unwrap_or(config.eth_start_from_block)` should be enough